testresultparser/parser_test.go

```diff
@@ -0,0 +1,98 @@
+				expectedResult := &Result{TestSuite: "com.udacity.gradle.test.PersonTest", Tests: 1, Failures: 1, Errors: 0, Skipped: 0, Time: "0.003", Failure: f}
+				Expect(Parse(getFilepath("gradle_failure_test.xml"), "default")).To(Equal(expectedResult))
+			})
+		})
+
+	})
+	Describe("Parsing TXT file", func() {
+		Context("With Maven-surefire plugin and TestNG", func() {
+			It("should return output struct", func() {
+				f := []Failure{{TestCase: "foo", Type: "java.lang.AssertionError", Message: "expected [foo] but found [1.0]"}}
+				expectedResult := &Result{TestSuite: "testNgMavenExample1.TestNgMavenExampleTest", Tests: 2, Failures: 1, Errors: 0, Skipped: 0, Time: "0.785", Failure: f}
+				Expect(Parse(getFilepath("testNG_failure_test.txt"), "surefire")).To(Equal(expectedResult))
+			})
+		})
+		Context("With Maven-surefire plugin and TestNG and multiple failures", func() {
+			It("should return output struct", func() {
+				f := []Failure{{
+					TestCase: "bar",
+					Type:     "java.lang.AssertionError",
+					Message:  "expected [foo] but found [2.0]",
+				},
+					{
+						TestCase: "foo",
+						Type:     "java.lang.AssertionError",
+						Message:  "expected [foo] but found [1.0]",
+					},
+				}
+				expectedResult := &Result{TestSuite: "testNgMavenExample1.TestNgMavenExampleTest", Tests: 4, Failures: 2, Errors: 0, Skipped: 1, Time: "1.395", Failure: f}
+				Expect(Parse(getFilepath("testNG_failure_test2.txt"), "surefire")).To(Equal(expectedResult))
+			})
+		})
+	})
+})
+
+// --- Test helpers
+
+func getFilepath(filename string) string {
+	_, packagefilename, _, _ := runtime.Caller(0)
+	targetDir := filepath.FromSlash(path.Dir(packagefilename) + "/artifacts/")
```

Comment:
I think `"/artifacts/"` is a bit too generic. I would rename the folder to `test_fixtures` instead.